src/item.cpp

```diff
@@ -3998,20 +3992,33 @@ int item::gun_pierce( bool with_ammo ) const
     return ret;
 }
 
-int item::gun_recoil( bool with_ammo ) const
+int item::gun_handling() const
 {
     if( !is_gun() ) {
         return 0;
+
     }
-    int ret = type->gun->recoil;
-    if( with_ammo && ammo_data() ) {
-        ret += ammo_data()->ammo->recoil;
-    }
+    double handling = type->gun->handling;
     for( const auto mod : gunmods() ) {
-        ret += mod->type->gunmod->recoil;
+        handling += mod->type->gunmod->handling;
     }
-    ret += damage() * 15;
-    return ret;
+
+    return std::max( pow( ( weight() / 250 ), 0.8 ) * pow( handling / 10.0, 1.2 ), 0.0 );
```

Comment:
For example of what I was talking about:
Consider a technomagic gun that has low weight, but handles perfectly due to handwave field stabilizing it. Then someone slaps a lead block/uranium battery on it as a mod. Now it handles incredibly well.

Plus, it means that adding useless mods gives a small bonus to handling. This isn't good.
Having only the base weight count would be a tiny bit less useless in the formula itself, but it would make more sense by not making people weight their guns with random mods for the purpose of improving handling.

Lightweight furniture isn't a good mod. It competes with more useful ones and only really makes sense for hollow bones mutants who have tiny carrying capacity.

I don't see a good reason to go with basic physics here. It all sounds like gameplay vs realism.